### PR TITLE
Add a function to set a object visit by given userIDs

### PR DESCRIPTION
**Pull request by Cyperghost**

### CODE REVIEW

File: wcfsetup/install/files/lib/system/visitTracker/VisitTracker.class.php [hunk 1]
```diff
@@ -149,8 +149,29 @@ public function deleteObjectVisits($objectType) {
 	}
 	
 	/**
+	 * Tracks an object visit by given userIds
+	 *
+	 * @param string    $objectType
+	 * @param integer   $objectID
+	 * @param integer[] $userIDs
+	 * @param int       $time
```
Comment (Gravatronics) [inline wcfsetup/install/files/lib/system/visitTracker/VisitTracker.class.php]: Please use indentation consistent to the other methods in this class.

File: wcfsetup/install/files/lib/system/visitTracker/VisitTracker.class.php [hunk 2]
```diff
@@ -149,8 +149,29 @@ public function deleteObjectVisits($objectType) {
...
+	 */
+	public function trackObjectVisitByUserIDs($objectType, $objectID, array $userIDs, $time = TIME_NOW) {
+		// save visit
+		$sql = "REPLACE INTO	wcf".WCF_N."_tracked_visit
+						(objectTypeID, objectID, userID, visitTime)
+				VALUES		(?, ?, ?, ?)";
```
Comment (Gravatronics) [inline wcfsetup/install/files/lib/system/visitTracker/VisitTracker.class.php]: Please fix the indentation of the second and third line.

File: wcfsetup/install/files/lib/system/visitTracker/VisitTracker.class.php [hunk 3]
```diff
@@ -149,8 +149,29 @@ public function deleteObjectVisits($objectType) {
...
+		$statement = WCF::getDB()->prepareStatement($sql);
+		$objectTypeID = $this->getObjectTypeID($objectType);
+		
+		foreach ($userIDs as $userID) {
+			$statement->execute([$objectTypeID, $objectID, $userID, $time]);
+		}
```
Comment (Gravatronics) [inline wcfsetup/install/files/lib/system/visitTracker/VisitTracker.class.php]: Please wrap this into a transaction.

File: wcfsetup/install/files/lib/system/visitTracker/VisitTracker.class.php [hunk 4]
```diff
@@ -149,8 +149,29 @@ public function deleteObjectVisits($objectType) {
...
+	}
+	
+	/**
 	 * Tracks an object visit.
-	 * 
+	 *
```
Comment (Gravatronics) [inline wcfsetup/install/files/lib/system/visitTracker/VisitTracker.class.php]: Please restore the whitespace (same below).

File: wcfsetup/install/files/lib/system/visitTracker/VisitTracker.class.php [hunk 5]
```diff
@@ -149,8 +149,29 @@ public function deleteObjectVisits($objectType) {
 	}
 	
 	/**
+	 * Tracks an object visit by given userIds
```
Comment (Gravatronics) [inline wcfsetup/install/files/lib/system/visitTracker/VisitTracker.class.php]: `Tracks an object visit for the users with the given ids.`

**Comment by dtdesign** [comment]
What is the motivation behind this?

**Comment by Cyperghost** [comment]
I need this function for my plugin, to replace the visit time by multi user

**Pull request merged**